pkgs/build-support/mkshell/default.nix

```diff
@@ -41,10 +41,14 @@ stdenv.mkDerivation ({
   shellHook = lib.concatStringsSep "\n" (lib.catAttrs "shellHook"
     (lib.reverseList inputsFrom ++ [ attrs ]));
 
-  nobuildPhase = ''
-    echo
-    echo "This derivation is not meant to be built, aborting";
-    echo
-    exit 1
+  phases = [ "buildPhase" ];
+
+  buildPhase = ''
+    echo "-------------------------------------------" >> $out
+    echo " Do not use this file. See the mkShell doc." >>$out
```

Comment:
I don't quite understand why.

Reply:
Ok, this is ready for another round of review.